Sources/SwiftParser/Lexer.swift

```diff
@@ -310,6 +303,91 @@ extension Lexer {
   }
 }
 
+// MARK: - Peeking
+
+extension Lexer.Cursor {
+  func peek(at offset: Int = 0) -> UInt8? {
+    assert(offset >= 0)
+    guard offset < self.input.count else {
+      return nil
+    }
+    return self.input[offset]
+  }
+
+  /// Essentially a UInt8 that is expressible by integer and string literals.
+  struct PeekMatch: ExpressibleByUnicodeScalarLiteral, ExpressibleByIntegerLiteral {
+    let value: UInt8
+
+    init(unicodeScalarLiteral value: Unicode.Scalar) {
+      self.value = UInt8(ascii: Unicode.Scalar(unicodeScalarLiteral: value))
+    }
+
+    init(integerLiteral value: UInt8) {
+      self.value = value
+    }
+  }
+
+  // MARK: Positive matches
+
+  /// Stamped out copy of `peek(at:matches:)` in case we are matching a single
+  /// character to avoid construction of an array.
+  func peek(at offset: Int = 0, matches character: PeekMatch) -> Bool {
```

Comment:
`advance` has `matching` which IMO reads better. You could also make `doesntMatch` `unless` instead. It sort of sounds like a condition, but not as much as `if` and `when` would for the positive case 🤷.

Reply:
Discussed offline to rename this to `is(at:)`